test/sub.mk

```diff
@@ -1,15 +1,18 @@
 # The default target is `test`, which runs in 3 phases
 # 1. Remove the compile.stamp file if the compiler jar has changes
-# 2. Compile the .bal test cases into .ll files
-# 3. Compile, execute and check the output of .ll files
+# 2. If COMPILE_MODE is default: compile the .bal test cases into .ll files 
+# 	 Else: compile the .bal test cases into .o files (using the jni version of the compiler)
+# 3. Compile, execute and check the output of .ll/.o files
 # Phase 2 only updates .ll if they have changed.
 # Phases 2 and 3 use recursive invocations of make so dependencies are recalculated.
 # This is used in phases 1 and 2.
 # To run this, first navigate to out/<category> dir, then
-# You can do `make -f ../../sub.mk tdir=$(basename "$PWD") compile` to compile all changed test cases from .bal to .ll
-# You can do `make -f ../../sub.mk tdir=$(basename "$PWD") testll` to test the ll files.
+# You can do `make -f ../../sub.mk tdir=$(basename "$PWD") compile` to compile all changed test cases from .bal to .ll/.o files
+# You can do `make -f ../../sub.mk tdir=$(basename "$PWD") testIntermediate` to test the ll/o files.
 # Failing tests are listed in fail.txt
 COMPILER_JAR=../../../build/compiler/bin/nballerina.jar
+JNI_COMPILER_JAR=../../../testbuild/target/bin/nballerina.jar
+COMPILE_MODE ?= default
```

Comment:
Use a variable name and value that makes use of kind of output produced: .ll or .o

Reply:
fixed